### PR TITLE
feat(shell-proxy): support SOCKS protocol based on URI scheme

### PR DESCRIPTION
**Pull request by liby**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- Identify the proxy protocol passed to netcat based on URI schemes

## Other comments:

- No longer pass proxy protocol options uniformly as `connect`

### CODE REVIEW

**Comment by mcornella** [comment]
@septs can you take a look?

**Comment by liby** [comment]
@septs PTAL. 🙏

File: plugins/shell-proxy/ssh-proxy.py [hunk 1]
```diff
@@ -5,10 +5,16 @@
 import urllib.parse
 
 proxy = next(os.environ[_] for _ in ("HTTP_PROXY", "HTTPS_PROXY") if _ in os.environ)
+
+if urllib.parse.urlparse(proxy).scheme == "http":
+    proxy_protocol = "connect"
+else:
+    proxy_protocol = "5"
```
[the remark this answers is not in the document]
Comment (septs) [reply plugins/shell-proxy/ssh-proxy.py]: if proxy url is `https://...`

i think can define a mapping

```python
protocols = {
    "http": "connect",
    "https": "connect",
    "socks": "5",
    "socks5": "5",
    "socks4": "4",
    "socks4a": "4",
}

if parsed.scheme not in protocols:
    raise Error("unsupported proxy protocol?") 

protocols[parsed.scheme]
```

**Comment by septs** [comment]
plz wait for oh-my-zsh staff approve.

**Comment by mcornella** [comment]
Thank you both!

**Pull request merged**